### DISCUSSION TITLE
Re-design 'k2s system ssh' and 'k2s system scp' Commands

### DISCUSSION THREAD

**Discussion by jgriegershs** [Siemens-Healthineers/K2s#787] [Ideas]
### Problems
- Currently, the `k2s system ssh` and `k2s system scp` are tightly coupled to one control-plane (i.e. `kubemaster`) and a worker node in `multi-vm` setup. In the near future, *K2s* will support multiple worker nodes.
- The semantics of e.g. `k2s system ssh [m|w] [-- <remote-command>]` confuses users.

### Solution Proposal
Re-design the aforementioned commands to align with the new `k2s node` command group, e.g.:
- `k2s node connect --ip 172.19.1.100` --> this would create an interactive SSH-based session to the control-plane
- `k2s node exec --ip 172.19.1.100 -c "echo 'hello'"` --> this would run a batch command on the control-plane using SSH batch execution without interactivity
- `k2s node copy --ip 172.19.1.100 --source "c:\temp\my.file" --target "/tmp/"` --> this would copy to control-plane using SCP

To display *K8s* node information, a user can run `kgn` which results in outputs like:
```
NAME         STATUS   ROLES           AGE   VERSION   INTERNAL-IP    EXTERNAL-IP   OS-IMAGE                         KERNEL-VERSION          CONTAINER-RUNTIME
kubemaster   Ready    control-plane   19d   v1.29.5   172.19.1.100   <none>        Debian GNU/Linux 11 (bullseye)   5.10.0-32-cloud-amd64   cri-o://1.29.8
pc-12345     Ready    worker          21h   v1.29.5   172.20.1.2     <none>        Windows 10 Enterprise            10.0.19045.4894         containerd://1.7.17
```

In future implementations, we could add flags like `-n kubemaster` or `-r control-plane` as well to determine the target node.

**Comment by jgriegershs** [comment]
Thinking about the copy command, we might need to continue distinguishing between copying to and from, so we might end up with something like:
- `k2s node copy-to --ip 172.19.1.100 --source "c:\temp\my.file" --target "/tmp/"`
- `k2s node copy-from --ip 172.19.1.100 --source "/tmp/my.file" --target c:\temp\"`

What do you think?

**Comment by mahenarayan** [comment]
@jgriegershs Thank you for writing down the proposal. I agree to the proposal with some comments. 

`k2s node copy --ip 172.19.1.100 --source "c:\temp\my.file" --target "/tmp/"`
`k2s node copy --reverse --ip 172.19.1.100 --source "c:\temp\my.file" --target "/tmp/"`

Just to be consistent with other tools out there but not a show stopper.

**Comment by jgriegershs** [comment]
`k2s node copy` command is already implemented.

The other two commands are described in #854 and #855.
